### PR TITLE
Fix set_tag and remove_tag requests

### PR DESCRIPTION
**Pull request by zaucker**
No return values, exception handling, OpenAPI validation

### CODE REVIEW

File: lib/Agrammon/DB/Dataset.pm6 [hunk 1]
```diff
@@ -147,31 +163,51 @@ class Agrammon::DB::Dataset does Agrammon::DB {
         return self;
     }
 
-    method set-tag($tagName) {
+
+    #| Set tag on datasets.
+    method set-tag(@datasets!, $tag-name! --> Nil) {
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:168]: Redundant `!` (positional parameter are required by default). Funnily enough, we introduced an annotation to point that out in this week's Comma release. :-)

File: lib/Agrammon/DB/Dataset.pm6 [hunk 2]
```diff
@@ -147,31 +163,51 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
+                    die X::Agrammon::DB::Dataset::SetTagFailed.new(:$tag-name);
+                }
             }
         }
-        return self;
     }
 
-    method remove-tag($tagName) {
+    #| Remove tag from datasets.
+    method remove-tag(@datasets!, $tag-name! --> Nil) {
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:190]: Same here

**Review by jnthn — APPROVED** [review]

File: lib/Agrammon/DB/Dataset.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6]: Removed.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6]: Removed

**Pull request merged**